trust-store-trust-policy-specification.md

```diff
@@ -141,20 +141,109 @@ The implementation must allow the user to execute custom validations. These cust
 
-1. Get the signing algorithm(hash+encryption) from the signing identity and validate that the signing algorithm is valid and allow-listed.
-1. Get the public key from the signing identity and validate the artifact integrity using the public key and signing algorithm identified in the previous step.
-1. Get and validate TrustStore and TrustPolicy for correctness.
-1. Get the signing identity from the signed artifact and validate it against the identities configured in the trust store. The signing identity should match or leads to at least one of the trusted identities configured in the trust store.
-    1. If signing identity is certificate then validate that certificate and certificate-chain leads to self-signed root.
-1. Perform [artifact expiry](#artifact-expiry) validations based on trust policy.
-1. Perform [artifact revocation](#artifact-revocation) validations based on trust policy
-1. Perform extended validation(If any).
+- User has configured [trust store](#trust-store) and [trust policy](#trust-policy).
 
-Here is high level uml diagram for signature evaluation:
+### Steps
 
-![uml diagram showing signature evaluation](./media/trust-store-trust-policy-evaluation.svg)
+1. **Validate that the signature envelope format is supported.**
+    1. Parse the signature envelope content based on the signature envelope type specified in the signature artifact manifest.
+    1. Validate that the content type indicated by the `cty` property value of protected headers in the signature envelope is supported.
+1. **Validate the signature envelope integrity.**
+    1. Get the signing algorithm(hash+encryption) from the signing identity(such as a certificate) and validate that signing algorithm satisfies [algorithm requirements](./signature-specification.md#signature-algorithm-requirements)
+    1. Get the public key from the signing identity and validate the signature envelope integrity using the public key and signing algorithm identified in the previous step.
+1. **Validate the signature against trust policy and trust store.**
+    1. Using the `scope` configured in trust policies, get the applicable trust policy. (Implementations might have this value precomputed, added it for completeness)
+    1. For the applicable trust policy, **validate trust-store:**
+        1. Validate that the signing certificate and certificate chain found in the signature envelope leads to a root certificate and satisfies [certificate requirements](./signature-specification.md#certificate-requirements).
+        1. For all the trust-stores configured in applicable trust-policy perform the following steps.
+            1. Validate that certificate and certificate-chain lead to a trusted certificate configured in the `x509Certs` field of trust-store.
+            1. If above verification succeeds then continue to the next step else iterate over the next set of trust stores. If all of the trust stores have been evaluated then fail the signature validation and exit.
```

Comment:
"iterate over the next set of trust store" => "next trust store" or "next set of root in trust store"